consensus/reactor.go

```diff
@@ -998,7 +998,7 @@ func (ps *PeerState) SetHasProposal(proposal *types.Proposal) {
 	}
 
 	ps.PRS.ProposalBlockPartsHeader = proposal.BlockID.PartsHeader
-	ps.PRS.ProposalBlockParts = bits.NewBitArray(proposal.BlockID.PartsHeader.Total)
+	ps.PRS.ProposalBlockParts = bits.NewBitArray(int(proposal.BlockID.PartsHeader.Total))
```

Comment:
its either we make `NewBitArray` accept an int64 or allow for this to happen. 
 cc @melekes ?

Reply:
I think we should have a max for `PartsHeader.Total` in `ValidateBasic`. Then it should be fine.

Reply:
It seems that the total could be more strict with a uint64. do you know if this number is ever very large?

Reply:
made the change to int32, now its ready fro review 😄